larky/src/main/resources/vendor/OpenPGP/__init__.star

```diff
@@ -605,72 +624,52 @@ def _class_Message():
 
         def _get(i, *args, **kwargs):
             # Already parsed packets
-            print("current index: ", self.idx)
+            # print("__iter__ for ", self, " => idx is", i)
+            # print("start", state)
+            # stacktrace()
             if (
                 state['PACKETS_START']
-                and state["START_POS"] < len(self._packets_start)
+                # and state["START_POS"] < len(self._packets_start)
                 # self._packets_start
-                # and i < len(self._packets_start)
+                and i < len(state['PACKETS_START'])
                 # and ((state["START_POS"] - i) < len(self._packets_start))
             ):
-                print("I AM HERE?")
-                print(self._packets_start)
-                self.idx += 1
-                rval = Ok(self._packets_start[state["START_POS"]])
-                state["START_POS"] = state["START_POS"] + 1
+                # print(self._packets_start)
+                # rval = Ok(self._packets_start[state["START_POS"]])
+                rval = Ok(self._packets_start[i])
+                state["START_POS"] = i
+                # print("in _packets_start, idx:", i, "current pos:", state["START_POS"])
                 return rval
 
             if self._input:
+                # print("self._input is not None for ", type(self))
                 if self._input.has_more():
                     packet = Packet.parse(self._input)
-                    print(packet)
+                    # print("in __iter__, parsed packet:", packet)
+                    # print("in __iter__, input has more?", self._input.has_more())
                     if packet:
                         self._packets_start.append(packet)
-                        self.idx += 1
                         return Ok(packet)
                     else:
                         fail("OpenPGPException: Parsing is stuck")
-            print("where did i get here?", self.idx)
+            # print("where did i get here?", self.idx)
```

Comment:
We need to remove all the prints/comment-out prints

Reply:
I can do that now. Would you want that now?

Reply:
That's ok. We need a massive refactoring/cleaning up